### PR TITLE
ITl-5 Implement Search Functionality

### PR DESCRIPTION
**Pull request by tiborFulop123**

### CODE REVIEW

File: src/components/HomePage.vue [hunk 1]
```diff
@@ -20,17 +19,22 @@
...
   import toDoListPlaceholder from './../assets/toDoListPlaceholder.svg';
 
   //begin region Variables
-
+  const emit = defineEmits(['toDoSearch']);
   const toDos = ref([]);
 
+  const searchTitle = ref('');
+
+  const searchedToDo = computed(() => {
```
Comment (gerebdavid) [inline src/components/HomePage.vue]: const searchedToDo = computed(() => toDos.value.filter((item) => item.title.includes(searchTitle.value)));

File: src/components/ToDoListSearch.vue [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<template>
+  <div class="flex justify-between border-black border-[2px] rounded-lg">
+    <div class="flex m-2"><img :src="searchIcon" /></div>
+    <div class="">
+      <input type="text" class="h-10" placeholder="Search todos..." v-model="searchField" />
+    </div>
+
+    <div>
+      <button class="m-2 bg-black text-white rounded-full" @click="searchToDo">
```
Comment (gerebdavid) [inline src/components/ToDoListSearch.vue]: Can you use the baseButton here?

File: src/components/HomePage.vue [hunk 3]
```diff
@@ -57,5 +61,9 @@
     toDos.value = upDatedToDos;
   }
 
+  function searchToDo(searchText) {
+    console.log(searchText);
```
Comment (gerebdavid) [inline src/components/HomePage.vue]: Do you still need this console.log?

File: src/components/ToDoListSearch.vue [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        <p>Search</p>
+      </button>
+    </div>
+  </div>
+</template>
+
+<script setup>
+  import searchIcon from './../assets/searchIcon.svg';
+  import { ref } from 'vue';
```
Comment (gerebdavid) [inline src/components/ToDoListSearch.vue:15]: First imports from vue and the the icon

File: src/components/Header.vue [hunk 5]
```diff
@@ -30,6 +26,10 @@
   function addToDo() {
     emit('toDoAdded');
   }
+  function searchToDo(searchField) {
+    console.log(searchField);
```
Comment (kincsokiss) [inline src/components/Header.vue]: Please add descriptive log message here, not just the value, or remove this line.

**Review by kincsokiss — CHANGES_REQUESTED** [review]

**Review by gerebdavid — APPROVED** [review]

**Review by roli-wg — APPROVED** [review]

**Review by komaromiGyorgy — APPROVED** [review]

**Review by kincsokiss — APPROVED** [review]

**Pull request merged**